src/basecamp.cpp

```diff
@@ -328,51 +360,126 @@ void basecamp::set_name( const std::string &new_name )
     name = new_name;
 }
 
-void basecamp::consume_components( inventory &camp_inv, const recipe &making, int batch_size,
+/*
+ * we could put this logic in map::use_charges() the way the vehicle code does, but I think
+ * that's sloppy
+ */
+long basecamp::use_charges( const itype_id fake_id, long quantity, std::list<item> &ret )
```

Comment:
`fake_id` could be passed by reference.

Also, is there a good reason not to follow the convention set by `map::use_charges()` of passing `quantity` by reference and returning an `std::list<item>`?

Reply:
Yes, the `use_charges_from_furn()` syntax that doesn't require a splice() is better than the `use_charges_from_stack()` syntax that does.

But it doesn't matter in this instance so I'll return the list.